packages/manager/src/features/PlacementGroups/PlacementGroupsCreateDrawer.tsx

```diff
@@ -49,13 +49,15 @@ export const PlacementGroupsCreateDrawer = (
     setHasRegionReachedPGCapacity,
   ] = React.useState<boolean>(false);
 
+  const [regionSelected, setRegionSelected] = React.useState<Region>();
+
   const selectedRegionFromProps = regions?.find(
     (r) => r.id === selectedRegionId
   );
```

Comment:
I'm noticing that we have two sources of truth for the selected region, one is the formik `values.region` and the other is `regionSelected`. This means that they both have to be updated separately (e.g., on lines 60 and 61).

We could make the formik `values.region` the single source of truth as below (you would have to move this useMemo to below the formik declaration):

```suggestion
  const selectedRegion = React.useMemo(
    () => regions?.find((region) => region.id == values.region),
    [regions, values.region]
  );
```

Then use `selectedRegion` in place of `regionSelected` and `selectedRegionFromProps`.

Reply:
@hkhalil-akamai Thank you for the suggestion, I went ahead and ran with it and things seem to be working fine and code is more maintainable this way.